specification/attributes/discount_handling.md

```diff
@@ -34,9 +34,10 @@ Indicates how to include and apply discounts to usage charges or rows.
 * Purchased discounts (e.g., commitment-based discounts) MUST be amortized.
   * The BilledCost MUST be 0 for any row where the commitment covers the entire cost for the charge period.
   * The EffectiveCost MUST include the portion of the amortized purchase cost that applies to this row.
-  * CommitmentDiscountStatus MUST be "Used" for rows that received a reduced price from that commitment.
-  * If a commitment is not fully utilized, the provider MUST include a row that represents the unused portion of the commitment for that charge period. CommitmentDiscountStatus MUST be "Unused".
   * The sum of the EffectiveCost for all rows where CommitmentDiscountStatus is "Used" or "Unused" for each CommitmentDiscountId over the entire duration of the commitment MUST be the same as the total BilledCost of the commitment-based discount.
+  * The CommitmentDiscountId and ResourceId MUST be set to the ID assigned to the commitment-based discount. ChargeCategory MUST be set to "Purchase" on rows that represent a purchase of a commitment-based discount.
```

Comment:
I'm not entirely sure if I'll manage to describe the use case we discussed yesterday, but here it is: 

What if a provider is unable to provide Purchase records containing IDs of Reserved Instance (RI) instances (assuming those are the commitment IDs we're seeking and would be provided in Usage records with commitment discounts applied), but only a Purchase record with the original Order ID (which allows retrieval of corresponding RI instance IDs through an API)? 
Would it be acceptable to change MUST' to SHOULD, at least for the time being? 
If not, what do we expect in  CommitmentId and ResourceId columns? null? while that OrderId would still be provided in one of the native columns?

Reply:
I thought @flanakin was ok with the ID assigned to the discount (which might be ok to be order-id for now? but obviously as we all agreed it would be best if it mapped to the actual discount?). I think it would be weird to say you don't have to provide the ID of the thing you bought - because that's what any consumer of the data would want to map/correlate to the usage records (amortization/coverage etc.). Open to ideas but I do think the language can't be so vague that any ID would meet the requirement.

Reply:
We don't have an instance ID at the time of purchase. We only have the order ID. They are effectively bulk purchases that result in a single charge.